src/java.base/share/classes/java/util/Base64.java

```diff
@@ -749,23 +810,19 @@ private int decode0(byte[] src, int sp, int sl, byte[] dst) {
 
             while (sp < sl) {
                 if (shiftto == 18 && sp + 4 < sl) {       // fast path
```

Comment:
Please change to sp < s1 - 4. Current version is sensitive to integer overflow. That's not a real problem in the current code, because the next check catches that, but we should better avoid this with the new intrinsics.

Reply:
Good catch. Will fix.